tests/src/test/scala/com/nvidia/spark/rapids/unit/StringRepeatUnitTest.scala

```diff
@@ -0,0 +1,108 @@
+
+import org.apache.spark.sql.rapids.GpuStringRepeat
+import org.apache.spark.sql.types.DataTypes
+
+// This class just covers some test cases that are not possible to call in PySpark tests.
+// The remaining tests are in `integration_test/src/main/python/string_test.py`.
+class StringRepeatUnitTest extends GpuUnitTests {
+  test("Test StringRepeat with scalar string and scalar repeatTimes") {
+    // Test repeat(NULL|str, NULL).
+    withResource(GpuScalar(null, DataTypes.IntegerType)) { nullInt =>
+      val doTest = (strScalar: GpuScalar) => {
+        withResource(GpuStringRepeat(null, null).doColumnar(1, strScalar, nullInt)) { result =>
+          assertResult(1)(result.getRowCount)
+          assertResult(result.getNullCount)(result.getRowCount)
+        }
+      }
+
+      withResource(GpuScalar(null, DataTypes.StringType)) { nullStr => doTest(nullStr) }
+      withResource(GpuScalar("abc123", DataTypes.StringType)) { str => doTest(str) }
+      withResource(GpuScalar("á é í", DataTypes.StringType)) { str => doTest(str) }
+    }
+
+    // Test repeat(NULL, intVal).
+    withResource(GpuScalar(null, DataTypes.StringType)) { nullStr =>
+      val doTest = (intVal: Int) =>
+        withResource(GpuScalar(intVal, DataTypes.IntegerType)) { intScalar =>
+          withResource(GpuStringRepeat(null, null).doColumnar(1, nullStr, intScalar)) { result =>
+            assertResult(1)(result.getRowCount)
+            assertResult(result.getNullCount)(result.getRowCount)
+          }
+        }
+
+      doTest(-1)
+      doTest(0)
+      doTest(1)
+    }
+
+    // Test repeat(str, intVal).
+    withResource(GpuScalar("abc123", DataTypes.StringType)) { strScalar =>
+      val doTest = (intVal: Int, expectedStr: String) =>
+        withResource(GpuScalar(intVal, DataTypes.IntegerType)) { intScalar =>
+          withResource(GpuStringRepeat(null, null).doColumnar(1, strScalar, intScalar)) {
+            result =>
+              withResource(result.copyToHost()) { hostResult =>
+                assertResult(1)(hostResult.getRowCount)
+                assertResult(0)(hostResult.getNullCount)
+                assertResult(expectedStr)(hostResult.getJavaString(0))
+              }
+          }
+        }
+
+      doTest(-1, "")
+      doTest(0, "")
+      doTest(1, "abc123")
+      doTest(2, "abc123abc123")
+      doTest(3, "abc123abc123abc123")
+    }
+  }
+
+  test("Test StringRepeat with NULL scalar string and column repeatTimes") {
+    val intCol = ColumnVector.fromBoxedInts(-3, null, -1, 0, 1, 2, null)
+
+    withResource(GpuColumnVector.from(intCol, DataTypes.IntegerType)) { intGpuColumn =>
+      // Test repeat(NULL, intVal).
+      withResource(GpuScalar(null, DataTypes.StringType)) { nullStr =>
+        withResource(GpuStringRepeat(null, null).doColumnar(nullStr, intGpuColumn)) { result =>
+          assertResult(intGpuColumn.getRowCount)(result.getRowCount)
+          assertResult(result.getNullCount)(result.getRowCount)
```

Comment:
Nit: The expected value is the row count (which we already validated above) and the value being tested is the null count, so these should be flipped.
```suggestion
          assertResult(result.getRowCount)(result.getNullCount)
```